src/graphql/entries.ts

```diff
@@ -63,34 +67,90 @@ function updateRichTextField(
   locale: string
 ) {
   if (name in modified) {
+    if (!modified[name]) {
+      modified[name] = {};
+    }
     (modified[name] as { json: unknown }).json = update?.fields?.[name]?.[locale] ?? null;
   }
 }
 
-function updateMultiRefField(
-  modified: Record<string, unknown>,
-  update: EntryProps,
-  name: string,
-  locale: string
+function getContentTypenameFromEntityReferenceMap(
+  referenceMap: Map<string, EntryProps | AssetProps>,
+  entityId?: string
 ) {
-  if (name in modified) {
-    // Listen to sorting
-    (modified[`${name}Collection`] as { items: CollectionItem[] }).items.sort((a, b) => {
-      const aIndex = update?.fields?.[name]?.[locale].findIndex(
-        (item: CollectionItem) => item.sys.id === a.sys.id
-      );
-      const bIndex = update?.fields?.[name]?.[locale].findIndex(
-        (item: CollectionItem) => item.sys.id === b.sys.id
-      );
-      return aIndex - bIndex;
+  if (referenceMap && entityId) {
+    const entity = referenceMap.get(entityId);
+    if (entity) {
+      const contentTypeId = entity.sys.contentType?.sys.id;
+      const typename = contentTypeId.charAt(0).toUpperCase() + contentTypeId.slice(1);
+      return typename;
+    }
+  }
+}
+
+function updateReferenceField(
+  updatedReference: EntryProps & { __typename?: string },
+  entityReferenceMap: Map<string, EntryProps | AssetProps>
+) {
+  // if the reference was deleted return null
+  if (updatedReference === null) {
+    return null;
+  }
+
+  // it's already in graphql format so we can return
+  if (updatedReference && updatedReference.__typename) {
```

Comment:
```suggestion
  if (updatedReference.__typename) {
```

according to the types it can not be undefined, so this should be enough?